cas-server-support-osf/src/main/java/org/pac4j/oauth/profile/orcid/OrcidAttributesDefinition.java

```diff
@@ -0,0 +1,59 @@
+/*
+  Copyright 2012 - 2015 pac4j organization
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.pac4j.oauth.profile.orcid;
+
+import org.pac4j.core.profile.converter.Converters;
+import org.pac4j.oauth.profile.OAuthAttributesDefinition;
+
+/**
+ * This class defines the attributes of the {@link OrcidProfile}.
+ *
+ * @author Jens Tinglev
+ * @author Longze Chen
+ * @since 1.7.1
+ */
+public class OrcidAttributesDefinition extends OAuthAttributesDefinition {
+
+    public static final String XML_ORCID = "common:path";
+    public static final String XML_FIRST_NAME = "personal-details:given-names";
+    public static final String XML_FAMILY_NAME = "personal-details:family-name";
+    public static final String XML_URI = "common:uri";
+    public static final String XML_CREATION_METHOD = "history:creation-method";
+    public static final String XML_CLAIMED = "history:claimed";
+    public static final String XML_LOCALE = "preferences:locale";
+
+    public static final String ORCID = "path";
+    public static final String FIRST_NAME = "given-names";
+    public static final String FAMILY_NAME = "family-name";
+
+
+    public OrcidAttributesDefinition() {
+
+        // Definitions that are used when OrcidClient parses the XML response for attributes
+        addAttribute(XML_ORCID, Converters.stringConverter);
+        addAttribute(XML_FIRST_NAME, Converters.stringConverter);
+        addAttribute(XML_FAMILY_NAME, Converters.stringConverter);
+        addAttribute(XML_URI, Converters.stringConverter);
+        addAttribute(XML_CREATION_METHOD, Converters.stringConverter);
+        addAttribute(XML_CLAIMED, Converters.booleanConverter);
+        addAttribute(XML_LOCALE, Converters.localeConverter);
+
+        // Definitions that are used when OSF-compatible attributes are added to the profile
```

Comment:
The attributes that are released to OSF. They must be defined in `OrcidAttributesDefinition` in order to be added to the attribute list of the user's profile.